src/block/CopperDoor.php

```diff
@@ -0,0 +1,63 @@
+<?php
+
+/*
+ *
+ *  ____            _        _   __  __ _                  __  __ ____
+ * |  _ \ ___   ___| | _____| |_|  \/  (_)_ __   ___      |  \/  |  _ \
+ * | |_) / _ \ / __| |/ / _ \ __| |\/| | | '_ \ / _ \_____| |\/| | |_) |
+ * |  __/ (_) | (__|   <  __/ |_| |  | | | | | |  __/_____| |  | |  __/
+ * |_|   \___/ \___|_|\_\___|\__|_|  |_|_|_| |_|\___|     |_|  |_|_|
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * @author PocketMine Team
+ * @link http://www.pocketmine.net/
+ *
+ *
+ */
+
+declare(strict_types=1);
+
+namespace pocketmine\block;
+
+use pocketmine\block\utils\CopperTrait;
+use pocketmine\item\Item;
+use pocketmine\math\Facing;
+use pocketmine\math\Vector3;
+use pocketmine\player\Player;
+
+class CopperDoor extends Door{
+	use CopperTrait{
+		onInteract as onInteractCopper;
+	}
+
+	public function onInteract(Item $item, int $face, Vector3 $clickVector, ?Player $player = null, array &$returnedItems = []) : bool{
+		if ($player !== null && !$player->isSneaking()) {
+			return parent::onInteract($item, $face, $clickVector, $player, $returnedItems);
+		}
+
+		if ($this->onInteractCopper($item, $face, $clickVector, $player, $returnedItems)) {
+			//copy copper properties to other half
+			$other = $this->getSide($this->top ? Facing::DOWN : Facing::UP);
+			$world = $this->position->getWorld();
+			if ($other instanceof CopperDoor) {
+				$other->setOxidation($this->oxidation);
+				$other->setWaxed($this->waxed);
+				$world->setBlock($other->position, $other);
+			}
+			return true;
+		}
+
+		return false;
```

Comment:
The player will not be able to open the door if he presses the shift and there is no item in his hand. In vanilla it's possible. This code fixes this issue
```
		if($player->isSneaking() && $this->onInteractCopper($item, $face, $clickVector, $player, $returnedItems)){
			//copy copper properties to other half
			$other = $this->getSide($this->top ? Facing::DOWN : Facing::UP);
			$world = $this->position->getWorld();
			if($other instanceof CopperDoor){
				$other->setOxidation($this->oxidation);
				$other->setWaxed($this->waxed);
				$world->setBlock($other->position, $other);
			}
			return true;
		}elseif($player !== null){
			return parent::onInteract($item, $face, $clickVector, $player, $returnedItems);
		}

		return false;
```

Note: The onInteract function in this class is not called if the player holds an item in his hand with the shift key pressed. This means that he will not be able to make the door waxed.

Reply:
> The player will not be able to open the door if he presses the shift and there is no item in his hand

You are right